aeppl/opt.py

```diff
@@ -244,10 +244,20 @@ def naive_bcast_rv_lift(fgraph, node):
     ]
     bcasted_node = lifted_node.op.make_node(rng, size, dtype, *new_dist_params)
 
+    rv_map_feature = getattr(fgraph, "preserve_rv_mappings", None)
+
+    new_bcast_out = bcasted_node.outputs[1]
+
+    if rv_map_feature is not None and rv_var in rv_map_feature.rv_values:
+        val_var = rv_map_feature.rv_values[rv_var]
+        new_val_var = at.broadcast_to(val_var, tuple(bcast_shape))
+        rv_map_feature.rv_values[new_bcast_out] = new_val_var
```

Comment:
Won't this add an extra logp term? If I am parsing this correctly, in the `joint_logp` of your first new test case there would now be 3 terms corresponding to `(x_vv, new_value_var, y_vv)`, no?

Reply:
> Won't this add an extra logp term? If I am parsing this correctly, calling `joint_logp` on your first new test case would now sum 3 terms corresponding to `(x_vv, new_value_var, y_vv)`, no?

Have you tried it?  We can always check the sum in the tests.

Reply:
The first test case might actually add an extra term; I'll try it out.